### PR TITLE
Add table, album, preserve data and testing

### PR DESCRIPTION
**Pull request by byiringiroscar**
####  Create Option and testing
This PR is raised to implement the following features 🆗 :

- [x] Create Option 
- [x] Add testing method
- [x] Preserve data
- [x] Setup linters

### How should it be tested ⬇️ ⬇️ ⬇️ 

clone this repository
     
```
git clone https://github.com/ZewdieMc/catalog-of-my-things/
```
Navigate to folder
```
cd catalog-of-my-things
```
Switch to this branch option-album
```
git switch option-album
```
Run project 
create an object then run 
- open the command prompt and run :
```
ruby main.rb
```
Testing part
```
cd music_test
```
```
ruby genre_spec.rb
```
```
ruby music_spec.rb
```

### Thanks for reviewing my PR 😄

### CODE REVIEW

File: albumaction/albumact.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                file_content = File.read(file_path)
+                music_data = JSON.parse(file_content)
+                music_data.each do |album|
+                    puts "Published date: #{album['publish_date']} ---- archived: #{album['archived']} ----- on spotify : #{album['on_spotify']}"
+                  end
+            end
+        else
+            puts "No album available"
+        end
+    end
```
Comment (Besufekad-HAZ) [inline albumaction/albumact.rb:22]: @ZewdieMc Good job and nice implementation in Albumaction class 🎉

**Review by Besufekad-HAZ — APPROVED** [review]
## Hello @byiringiroscar   👏🏽, 

clapping

# ✔️✔️✔️   Status: APPROVED 🏆 🎉 🎊 

Your pull request is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Used correct git flow 👍🏽
- All added tests are passing👌🏽
- All linters are passing ❌ 
- Good work 👏🏽 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@Besufekad-HAZ ) in your question so I can receive the notification.

------

File: albumaction/albumact.rb (hunk 1 above)
Comment (Besufekad-HAZ) [reply albumaction/albumact.rb:22]: Hello @byiringiroscar can you please fix the linter and Rspec tests error?.....the conflicts has been already resolved so after fixing the error you can merge it. 😃

**Review by ZewdieMc — CHANGES_REQUESTED** [review]
# ❌ INVALID PULL REQUEST

- [ ] Kindly make sure that the tests are passed.
- [ ] Kindly fix the linter errors

File: albumaction/albumact.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
+require 'json'
+require_relative '../music'
+
+class Albumaction
+  def getmusicalbum
+    file_path = './album_data/music.json'
+    if File.exist?(file_path)
+      file_content = File.read(file_path)
+      JSON.parse(file_content)
```
Comment (ZewdieMc) [inline albumaction/albumact.rb:9]: [OPTIONAL] These two lines are already in line numbers 13 and 14, so they can be deleted

**Pull request merged**